packages/box/src/styles/rainbow-sprinkles.css.ts

```diff
@@ -0,0 +1,117 @@
+import { vars } from '@launchpad-ui/vars';
+import { flatten } from 'flat';
+import { defineProperties, createRainbowSprinkles } from 'rainbow-sprinkles';
+
+const { bg, border, fill, shadow, text, gradient, ...global } = vars.color;
+
+type FlattenObjectKeys<T extends Record<string, unknown>, Key = keyof T> = Key extends string
+  ? T[Key] extends Record<string, unknown>
+    ? `${Key}.${FlattenObjectKeys<T[Key]>}`
+    : `${Key}`
+  : never;
+
+type GlobalKeys = FlattenObjectKeys<typeof global>;
+type GradientKeys = FlattenObjectKeys<typeof gradient>;
+type BackgroundKeys = FlattenObjectKeys<typeof bg>;
+type BorderKeys = FlattenObjectKeys<typeof border>;
+type FillKeys = FlattenObjectKeys<typeof fill>;
+type TextKeys = FlattenObjectKeys<typeof text>;
+
+const colors = flatten<typeof global, Record<GlobalKeys, string>>(global);
```

Comment:
A question about how we tokenize these. Do we want to use hyphen `-` delimiters rather than periods `.`? That would map more cleanly to how they're expressed in CSS.

Reply:
Cool, that makes sense to me!